agent/engine/docker_task_engine.go

```diff
@@ -1336,25 +1339,55 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 
 		containerCredSpec, err := container.GetCredentialSpec()
 		if err == nil && containerCredSpec != "" {
-			// CredentialSpec mapping: input := credentialspec:file://test.json, output := credentialspec=file://test.json
+			// on windows CredentialSpec mapping: input := credentialspec:file://test.json, output := credentialspec=file://test.json
+			// on linux CredentialSpec mapping: input := ssm/asm arn, output := /var/credentials-fetcher/krbdir/123456/ccname_webapp01_xyz
 			desiredCredSpecInjection, err := credSpecResource.GetTargetMapping(containerCredSpec)
-			if err != nil || desiredCredSpecInjection == "" {
-				missingErr := &apierrors.DockerClientConfigError{Msg: "unable to fetch valid credentialspec mapping"}
-				return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(missingErr)}
-			}
+			if config.OSType == "windows" {
```

Comment:
In regards to a previous comment, I meant factoring this bit out into a function, with the "windows" function living in https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_task_engine_windows.go and the "linux" function living in https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_task_engine_linux.go

then you don't have to switch on OSType or runtime.GOOS, you can just call the function and the build flag means the correct one is used.

Reply:
sure will update accordingly

Reply:
updated PR